internal/local/k8s/provider.go

```diff
@@ -0,0 +1,91 @@
+package k8s
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	DockerDesktop = "docker-desktop"
+	Kind          = "kind"
+	Test          = "test"
+)
+
+// Provider represents a k8s provider.
+type Provider struct {
+	// Name of this provider
+	Name string
+	// ClusterName is the name of the cluster this provider will interact with
+	ClusterName string
+	// Context this provider should use
+	Context string
+	// Kubeconfig location
+	Kubeconfig string
+	// HelmNginx additional helm values to pass to the nginx chart
+	HelmNginx []string
+}
+
+// MkDirs creates the directories for this providers kubeconfig.
+// The kubeconfigs are always scoped to the user's home directory.
+// TODO: rename to something more clear
+func (p Provider) MkDirs(userHome string) error {
+	const permissions = 0700
+	kubeconfig := filepath.Join(userHome, p.Kubeconfig)
+	if err := os.MkdirAll(filepath.Dir(kubeconfig), permissions); err != nil {
+		return fmt.Errorf("could not create directory %s: %v", kubeconfig, err)
+	}
+
+	return nil
+}
+
+var (
+	// DockerDesktopProvider represents the docker-desktop provider.
+	DockerDesktopProvider = Provider{
```

Comment:
Suggestion: Consider moving these to a `map[string]Provider` this way in any error handling you can and enumerate the valid options

Reply:
Agreed.  Will revisit this in a follow-up PR.